File: A_Zhan_s_Blender.py
def main():
    import sys
    input = sys.stdin.read
    
    data = input().splitlines()  # Read all input at once
    t = int(data[0])  # First line contains the number of test cases
    results = []

    index = 1
    for _ in range(t):
        n = int(data[index])  # Number of fruits
        index += 1
        x, y = map(int, data[index].split())  # Blender capacity and input rate
        index += 1

        if n == 0:
            results.append(0)  # If no fruits, time is 0
            continue

        # Calculate minimum time needed
        min_time = (n + y - 1) // y  # Time to fill the blender
        blended_time = (n + x - 1) // x  # Time to blend all fruits

        # Append the maximum of the two times to results
        results.append(max(min_time, blended_time))

    # Output all results
    for result in results:
        print(result)

if __name__ == '__main__':
    main()

File: A_Zhan_s_Blender.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class A_Zhan_s_Blender {
    public static void main(String[] args) {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        try {
            int t = Integer.parseInt(reader.readLine()); // Number of test cases
            StringBuilder result = new StringBuilder();

            for (int i = 0; i < t; i++) {
                long n = Long.parseLong(reader.readLine()); // Number of fruits
                String[] inputs = reader.readLine().split(" ");
                long x = Long.parseLong(inputs[0]); // Blender capacity
                long y = Long.parseLong(inputs[1]); // Input rate

                // If there are no fruits, print 0
                if (n == 0) {
                    result.append(0).append("\n");
                    continue;
                }

                // Calculate minimum time needed
                long minTime = (n + y - 1) / y; // Time to fill the blender
                long blendedTime = (n + x - 1) / x; // Time to blend all fruits

                // Append the maximum of the two times to the result
                result.append(Math.max(minTime, blendedTime)).append("\n");
            }

            // Output all results
            System.out.print(result);
        } catch (IOException e) {
            e.printStackTrace(); // Handle IO exceptions
        }
    }
}
